src/components/SingleTopic.jsx

```diff
@@ -0,0 +1,49 @@
+import { useEffect, useState } from "react";
+import { Link, useParams } from "react-router-dom";
+import { getArticlesByTopic } from "../api";
+import LoadingSpinner from "../loading";
+
+export const SingleTopic = () => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [topic, setTopic] = useState([]);
+  const { topic_slug } = useParams();
+  useEffect(() => {
+    setIsLoading(true);
+    getArticlesByTopic(topic_slug).then((topicData) => {
+      console.log(topicData);
+      setIsLoading(false);
+      setTopic(topicData);
+    });
+  }, [topic_slug]);
+
+  return (
+    <div>
+      <h2>Articles</h2>
+      <fieldset>
+        <legend>Topics</legend>
+        <Link to="/articles">All</Link>
+        <Link to="/topics/coding">Coding</Link>
+        <Link to="/topics/cooking">Cooking</Link>
+        <Link to="/topics/football">Football</Link>
```

Comment:
right now the topics are hard coded.  what happens if in the future another category is added?

we want to get topics dynamically by fetching them from the api (GET /api/topics) and use the results of that to then create these links.

Reply:
It looks as though you still need to find a way of displaying these links dynamically (and in your AllArticles component, perhaps you need to extract it into its own component to DRY up your code. 
You could getTopics from your api.. then map over the response to create links that would remove  the need to have these hardcoded topics